### ISSUE TITLE
Linter rules for Navigator.method(..) vs Navigator.of(..).method()

### ISSUE THREAD

**Issue by themisir** [dart-lang/linter#2892]
**Describe the rule you'd like to see implemented**
Usually in flutter projects It's recommended to use one of the conventions for calling navigator methods:

1. `Navigator.of(context).pop();`
2. `Navigator.pop(context);`

I think it would be nice if dart had a linter to lint mix of 2 conventions. Because it's very easy to sometimes forget the project wide conventions or new members might not notice the convention. 

**Examples**

BAD:

```dart
Navigator.of(context).push(MaterialRoute(          // uses Navigator.of convention
  builder: (context) => MyWidget(
    onClick: () => Navigator.pop(context);         // uses Navigator.method convention
  ),
));
```

GOOD:

```dart
Navigator.of(context).push(MaterialRoute(          // uses Navigator.of convention
  builder: (context) => MyWidget(
    onClick: () => Navigator.of(context).pop();    // uses Navigator.method convention
  ),
));
```
